radicle-cob/src/type_name.rs

```diff
@@ -0,0 +1,78 @@
+// Copyright © 2022 The Radicle Link Contributors
+//
+// This file is part of radicle-link, distributed under the GPLv3 with Radicle
+// Linking Exception. For full terms see the included LICENSE file.
+
+use std::{fmt, str::FromStr};
+
+use serde::{Deserialize, Serialize};
+use thiserror::Error;
+
+/// The typename of an object. Valid typenames MUST be sequences of
+/// alphanumeric characters separated by a period. The name must start
+/// and end with an alphanumeric character
+///
+/// # Examples
+///
+/// * `abc.def`
+/// * `xyz.rad.issues`
+/// * `xyz.rad.patches.releases`
+#[derive(Clone, Debug, Eq, PartialEq, PartialOrd, Ord, Hash, Serialize, Deserialize)]
+pub struct TypeName(String);
+
+impl TypeName {
+    /// A string representation of the typename which will match the typename in
+    /// regular expressions. This primarily escapes periods
+    pub fn regex_safe_string(&self) -> String {
+        self.0.replace('.', "\\.")
+    }
```

Comment:
Are we still using this after removing regex?

Reply:
Probably not! Can try remove and see what happens